src/_blog/webtransport.md

```diff
@@ -0,0 +1,248 @@
+
+`/ip4/1.2.3.4/udp/4001/quic/webtransport/certhash/<hash>`.
+
+The `certhash` component of the multiaddress tells the browser the certificate hash, allowing it to establish the WebTransport connection successfully.
+
+In practice, you’ll see addresses containing multiple certificate hashes, e.g.
+
+`/ip4/1.2.3.4/udp/4001/quic/webtransport/certhash/<hash1>/certhash/<hash2>`.
+
+As described above, option (2) is intended for short-lived deployments, and browsers will only accept valid certificates for less than 14 days. We work around this constraint by generating two certificates: one valid for immediate use and another valid from when the first one expires. After 14 days, the server can forward its certificates and advertise an updated address containing the new certificate hash to the network.
+
+### Deep dive: How WebTransport works
+
+Let’s dive into the details! Understanding this section is unnecessary if you want to use WebTransport, so feel free to skip ahead.
+
+
+<div class="container" style="display:flex; column-gap:10px;">
+    <figure>
+        <img src="../assets/webtransport_diagram.jpg" >
+        <figcaption style="font-size:x-small;">
+        </figcaption>
+    </figure>
+</div>
+
+1. The browser dials a regular HTTP/3 connection to the server, verifying the certificate either by its chain of trust or by the certificate's hash.
+2. The browser sends an `Extended CONNECT` request on an HTTP/3 stream, requesting establishing a WebTransport session. If the server sends a `200` HTTP status, the WebTransport session is successfully established.
+
+Both sides can now open streams (both bidirectional and unidirectional) and send (unreliable) HTTP datagrams.
+
+> Note: the unreliable datagrams are a property of UDP, which QUIC is built on top of. This unreliable and out-of-order delivery makes UDP, QUIC, and WebTransport perfect for applications like video streaming or browser gaming, where low latency and speed are paramount.
+
+In libp2p, we still need to verify the libp2p peer IDs, so we’re not entirely done yet.
+
+To do so, the browser opens a new WebTransport stream and starts a [Noise](https://noiseprotocol.org/noise.html) handshake. This is the same handshake we use to secure connections on top of TCP in libp2p. In this handshake, we bind the outer connection to this handshake, which allows us to verify that we’re not falling victim to a man-in-the-middle (MITM) attack.
+
+Therefore, setting up a WebTransport connection in libp2p takes no more than three network roundtrips. Compare that to the six roundtrips we needed for WebSocket!
+
+We might even be able to bring this down to 2 roundtrips in the future: In principle, it should be possible to run the `CONNECT` request and the Noise handshake in parallel. Currently, the browser API doesn’t allow this, but we’ve [submitted a proposal](https://github.com/w3c/webtransport/issues/435) to enable this feature.
+
+<!-- TODO: do we need another diagram for this? -->
```

Comment:
@marten-seemann can you make a sketch for this diagram and share with Crcle? the more diagrams the better

Reply:
Decided to publish this without the diagram (for now).